src/editor/EditorButton.svelte

```diff
@@ -0,0 +1,41 @@
+<script>
+    export let flat = false;
+    export let name = 'button';
+    
+    export let onclick;
+</script>
+
+<button on:click={onclick} class:flat >
```

Comment:
You can forward the click event by not specifying a function to run at `on:click` like so:

```suggestion
<button on:click class:flat >
```

You can then just watch for a click event when consuming the component:
```svelte
<EditorButton on:click={dostuff}>...</EditorButton>
```